classes/Wrapper/ProductWrapper.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * 2007-2020 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\Ps_Googleanalytics\Wrapper;
+
+use PrestaShop\Module\Ps_Googleanalytics\Hooks\WrapperInterface;
+
+class ProductWrapper implements WrapperInterface
+{
+    private $context;
+
+    public function __construct(\Context $context)
+    {
+        $this->context = $context;
+    }
+
+    /**
+     * wrap products to provide a standard products information for google analytics script
+     */
+    public function wrapProductList($products, $extras = [], $full = false)
+    {
+        $result_products = [];
+        if (!is_array($products)) {
+            return;
+        }
+
+        $currency = new \Currency($this->context->currency->id);
+        $usetax = (\Product::getTaxCalculationMethod((int) $this->context->customer->id) != PS_TAX_EXC);
+
+        if (count($products) > 20) {
```

Comment:
$full = (count($products) <= 20);

Reply:
Thanks for your review.
This part of the code comes from the older version of the module. I've just placed this script in new class in order to change the module's structure.
As I don't have such time to handle it, I do not plan to refacto the old code for now.